espresso/core/java/androidx/test/espresso/util/SpecialCharacter.java

```diff
@@ -0,0 +1,47 @@
+package androidx.test.espresso.util;
+
+import android.util.Pair;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.hamcrest.Description;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class SpecialCharacter {
+
+  private static final List<Pair<String, String>> SPECIAL_CHAR_CODES = new LinkedList<Pair<String, String>>() {{
```

Comment:
Why map is a better here? Is it just because of a natural representation of the mapping paradigm?

All we need here is an iterable collection. Imho, the map seems a bit clumsy in this case.

Reply:
Only if you keep the way you are replacing.

Your algorithm creates an iterator and then does a full-length replacement, which has the complexity of O(N*M), time and space. It creates intermediate objects on the heap.

What they suggest is to use a Map, and replace characters, a pseudo code:
```java
for (char c : text) {
    output.append(MAP.get(c) ?: c);
}
```
This version would run in O(N) because only once it has to go through the input and use O(N) storage too because of the buffer.

Tip: Look up double brace initializer (a code smell) and Big O Notation (for calculating complexity).

Reply:
Thank you for the thorough explanation of your idea.

Yes, I do know about Big O and its suboptimal complexity.
I just assume this code be running on small portions of data (both replacement and strings), on which this complexity won't be crucial, but its readability seems to be slightly higher.

Well, since they don't use in JB any multi-chars variants https://github.com/JetBrains/intellij-community/blob/master/platform/platform-impl/src/com/intellij/openapi/editor/impl/view/SpecialCharacterFragment.java#L19-L44, I might be gone the wrong road.

Fixed. 
@adazh @TWiStErRob Colleagues, thanks for the suggested solution.

Reply:
Howdy!
I'm currently located in Russia and expect github being inaccessible here soon :)
Considering this fact, I'd like to complete this PR's changes.

Kindly, check the latest changes, I hope it's ok now.